app/src/main/java/io/appium/uiautomator2/utils/Device.java

```diff
@@ -36,11 +36,20 @@ public static void wake() throws RemoteException {
         getUiDevice().wakeUp();
     }
 
-    public static void scrollTo(String scrollToString) throws UiObjectNotFoundException {
-        // TODO This logic needs to be changed according to the request body from the Driver
+    public static void scrollToElement(UiSelector selector, int maxSwipes) throws UiObjectNotFoundException {
         UiScrollable uiScrollable = new UiScrollable(new UiSelector().scrollable(true).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().descriptionContains(scrollToString).instance(0));
-        uiScrollable.scrollIntoView(new UiSelector().textContains(scrollToString).instance(0));
+
+        if(uiScrollable.getClassName().equals(android.widget.HorizontalScrollView.class.getName())) {
```

Comment:
Please format the code

Reply:
can .getClassName() return null?

Reply:
1. I assume, the problem with code-formatting was the line slightly longer than 100 characters. I'll fix this.
2. All UI elements seem to have their **className** property set, so it is highly unlikely that  method **getClassName()** will return null for **UiScrollable**. However, it won't hurt to add a check-for null here.

Reply:
> I assume, the problem with code-formatting was the line slightly longer than 100 characters. I'll fix this.

This one plus a space after `if`. Usually it is just enough to perform Cmd+Shift+L in IDEA

> All UI elements seem to have their className property set, so it is highly unlikely that method getClassName() will return null for UiScrollable. However, it won't hurt to add a check-for null here.

Feel free to use Objects.equal, which guarantees null-safery